### ISSUE TITLE
JUnit test failures for some architectures in Debians CI

### ISSUE THREAD

**Issue by vanitasvitae** [vanitasvitae/pgpainless#309]
There are some strange test failures in Debians CI run on different architectures:

* [RiscV64](https://ci.debian.net/packages/p/pgpainless/unstable/riscv64/):
  * InlineSIgnVerifyRoundtripTest:testInlineSignAndVerifyWithBinarySignatures()
  * InlineDetachTest:detachCleartextSignedMessage()
* [armel](https://ci.debian.net/packages/p/pgpainless/unstable/armel/):
  * GenerateKeyWithAdditionalUserIdTest:test()

Not sure what to make out of this. The armel fail could only be caused by a very slow machine, where generation of the key takes longer than a minute. Maybe its worthwhile increasing the expiration time?

The other failures are more complex, as they are related to signature verification.

Perhaps those tests are just a bit flaky and restarting them fixes the issue? In any case, we should investigate and mitigate any reliability issues.

**Comment by vanitasvitae** [comment]
As pointed out by @jcharaoui, those arches are a bit flaky, so nothing to be too worried about.

**Comment by vanitasvitae** [comment]
I created #311 which allows to set custom reference times when creating signatures on keys. That way we are more flexible when setting up tests which check signature interpretation based on reference times.

While doing this I also increased the size of time frames by a factor of 3600 (from seconds to hours), so slow machines/architectures *should* now have plenty of time to execute without the risk of running out of the tests nominal time frames.
